### PR TITLE
[PM-12592] & [PM-12593] bottom nav notification dot

### PR DESCRIPTION
**Pull request by dseverns-livefront**
## 🎟️ Tracking
https://bitwarden.atlassian.net/browse/PM-12592
https://bitwarden.atlassian.net/browse/PM-12593

## 📔 Objective
- show the count of "settings" notifications in a badge on the bottom nav bar.

## 📸 Screenshots

https://github.com/user-attachments/assets/2494f010-b63a-4092-b850-88d76bf9e8e0

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/vaultunlockednavbar/VaultUnlockedNavBarScreen.kt [hunk 1]
```diff
@@ -277,18 +280,30 @@ private fun VaultBottomAppBar(
...
-                            },
-                        ),
-                        contentDescription = stringResource(
-                            id = destination.contentDescriptionRes,
-                        ),
-                    )
+                    BadgedBox(
+                        badge = {
+                            this@BottomAppBar.AnimatedVisibility(
+                                visible = destination.notificationCount > 0,
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/vaultunlockednavbar/VaultUnlockedNavBarScreen.kt:286]: How does the `AnimatedVisibility` look here?

Mostly curious how using the `RowScope.AnimatedVisibility` interacts with this `BoxScope`.

File: app/src/main/java/com/x8bit/bitwarden/ui/platform/components/badge/NotificationBadge.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * Reusable component for displaying a notification badge.
+ *
+ * @param notificationCount optional value to display in center of the badge.
+ */
+@Composable
+fun NotificationBadge(
+    modifier: Modifier = Modifier,
+    notificationCount: Int = 0,
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/platform/components/badge/NotificationBadge.kt:25]: Can we drop the default on `notificationCount`, I think we want that passed in every time.

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/platform/components/badge/NotificationBadge.kt:25]: Sure!

File: app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/vaultunlockednavbar/VaultUnlockedNavBarScreen.kt (hunk 1 above)
Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/vaultunlockednavbar/VaultUnlockedNavBarScreen.kt:286]: Set up some code to remove the badge and this is the results, I did update the transitions as the default looked a bit janky for the row scope one ->

| explicit compose.animiation | rowScoped |
|--------|--------|
| https://github.com/user-attachments/assets/a4294982-c3d7-4128-9a08-da2cdb3a33e3 | https://github.com/user-attachments/assets/dc5206a6-1d09-4bfa-bd2f-683e05f3a675 |

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/vaultunlockednavbar/VaultUnlockedNavBarScreen.kt:286]: basically can call it like `androidx.compose.animation.AnimatedVisibility()` here or leave it as it is.

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/vaultunlockednavbar/VaultUnlockedNavBarScreen.kt:286]: I am going to lean toward the explicit call and place a comment just cause the crossing the Scope is kinda funky.

File: app/src/main/java/com/x8bit/bitwarden/ui/platform/components/badge/NotificationBadge.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+@Composable
+fun NotificationBadge(
+    modifier: Modifier = Modifier,
+    notificationCount: Int,
+    backgroundColor: Color = LocalNonMaterialColors.current.fingerprint,
+    contentColor: Color = MaterialTheme.colorScheme.onSecondary,
+) {
+    Badge(
+        content = {
+            if (notificationCount > 0) {
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/platform/components/badge/NotificationBadge.kt:31]: Should the badge are if the number is 0 or below?

We could just remove this check, right?

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/platform/components/badge/NotificationBadge.kt:31]: sure I can agree as we have not design for the dot w/o the text rn.

Comment (dseverns-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/platform/components/badge/NotificationBadge.kt:31]: wait well would we then want to move this text up above the "Badge" and put the visibility inside of here? Thinking about this as a standalone component it doesn't have the knowledge of when it shows or not?

**Review by david-livefront — APPROVED** [review]

**Pull request merged**